### ISSUE TITLE
GC-safe sqlite3_destructor_type?

### ISSUE THREAD

**Issue by michaelsbradleyjr** [arnetheduck/nim-sqlite3-abi#4]
I ran into a problem with:
https://github.com/arnetheduck/nim-sqlite3-abi/blob/fda455cfea2df707dde052034411ce63de218453/sqlite3_abi.nim#L22-L23

In some circumstances the Nim compiler fails with an error message like:
```
/path/to/.nimble/pkgs/sqlite3_abi-3.34.0/sqlite3_abi.nim(23, 7) Error: 'myIter' is not GC-safe as it performs an indirect call here
```

One workaround is:
```nim
type
  my_sqlite3_destructor_type = proc (a1: pointer) {.cdecl, gcsafe, raises: [Defect].}

const
  MY_SQLITE_TRANSIENT = cast[my_sqlite3_destructor_type](-1)
```

Any thoughts or suggestions on the best way to approach this problem? Is there something that could be changed so this library's generated wrapper indicates `gcsafe` for `sqlite3_destructor_type`?

**Comment by arnetheduck** [comment]
indeed, all exported `proc` types should be both `raises: []` and `gcsafe`, similar to https://github.com/status-im/nim-bearssl/blob/master/bearssl/abi/bearssl_aead.nim#L4 - not sure how to best achieve that with nimterop, though it should be possible somehow.
